tests/models/beit3/test_modeling_beit3.py

```diff
@@ -0,0 +1,613 @@
+        model = Beit3ForVisualReasoning.from_pretrained("Raghavan/beit3_base_patch16_224_nlvr2").to(torch_device)
+        tokenizer = XLMRobertaTokenizer.from_pretrained("Raghavan/beit3_base_patch16_224_nlvr2")
+
+        image_processor = BeitImageProcessor.from_pretrained("Raghavan/beit3_base_patch16_224_nlvr2")
+        image = prepare_img()
+        beit3_processor = Beit3Processor(image_processor, tokenizer)
+        input = beit3_processor(text=["This is photo of a cat"], images=image)
+
+        model.eval()
+        # prepare bool_masked_pos
+        pixel_values = torch.cat(
+            (torch.tensor(input["pixel_values"]).unsqueeze(1), torch.tensor(input["pixel_values"]).unsqueeze(1)), dim=1
+        )
+        # forward pass
+        output = model(
+            input_ids=torch.tensor(input["input_ids"]),
+            pixel_values=pixel_values,
+            attention_mask=torch.ones(input["input_ids"].shape),
+        )
+        assert output.logits.shape == torch.Size([1, 2])
+        torch.testing.assert_allclose(output.logits.detach(), torch.tensor([[6.593818, -6.582055]]))
+
+    @slow
+    def test_inference_beit3_for_image_captioning(self):
+        model = Beit3ForCaptioning.from_pretrained("Raghavan/beit3_base_patch16_480_coco_captioning").to(torch_device)
+        tokenizer = XLMRobertaTokenizer.from_pretrained("Raghavan/beit3_base_patch16_480_coco_captioning")
+
+        image_processor = BeitImageProcessor.from_pretrained("Raghavan/beit3_base_patch16_480_coco_captioning")
+        image = prepare_img()
+        beit3_processor = Beit3Processor(image_processor, tokenizer)
+        input = beit3_processor(text=["This is photo of a cat"], images=image)
+
+        model.eval()
+        # prepare bool_masked_pos
+        language_masked_pos = torch.zeros((input["input_ids"].shape[0], input["input_ids"].shape[1]))
+        language_masked_pos[0, 5] = 1
+        input_tokens = list(input["input_ids"][0])
+        input_tokens[5] = 64001
+        output = model(
+            input_ids=torch.tensor([input_tokens]),
+            pixel_values=torch.tensor(input["pixel_values"]),
+            attention_mask=torch.zeros(language_masked_pos.shape),
+            language_masked_pos=language_masked_pos,
+        )
+        assert output.logits.shape == torch.Size([1, 64010])
+        torch.testing.assert_allclose(output.logits.detach()[0, :3], torch.tensor([-2.697914, -2.697912, -2.645459]))
+
+    @slow
+    def test_inference_beit3_for_image_text_retrieval(self):
+        model = Beit3ForImageTextRetrieval.from_pretrained("Raghavan/beit3_base_patch16_384_coco_retrieval").to(
+            torch_device
+        )
+        tokenizer = XLMRobertaTokenizer.from_pretrained("Raghavan/beit3_base_patch16_384_coco_retrieval")
+
+        image_processor = BeitImageProcessor.from_pretrained("Raghavan/beit3_base_patch16_384_coco_retrieval")
+        image = prepare_img()
+        beit3_processor = Beit3Processor(image_processor, tokenizer)
```

Comment:
this should be updated to just do `processor = Beit3Processor.from_pretrained(...)`, which includes the image processor and tokenizer